### PR TITLE
0.2/218 clear failed voucher codes on submit

### PR DESCRIPTION
**Pull request by charlesstrange2**

### CODE REVIEW

File: src/pages/Payment.vue [hunk 1]
```diff
@@ -84,11 +84,11 @@ export default {
         onRequestPayment() {
             Store.pendRecVouchers(
                 // on Success, route to /account
-                function() {
+                function(response) {
```
Comment (katjam) [inline src/pages/Payment.vue:87]: You need?

File: src/pages/Payment.vue [hunk 2]
```diff
@@ -84,11 +84,11 @@ export default {
...
                     this.$router.push('/account');
                 }.bind(this),
                 // on Failure... hook for an alert?
-                function() {
+                function(error) {
```
Comment (katjam) [inline src/pages/Payment.vue:91]: And this?

File: src/pages/Tap.vue [hunk 3]
```diff
@@ -60,12 +63,19 @@ export default {
             if (this.voucherCode !== null && this.voucherCode.length > 0) {
                 Store.addVoucherCode(this.sponsorCode.toUpperCase()+this.voucherCode,
                     // Success function
-                    function() {
+                    function(response) {
+                            // Add error message for invalid and fail codes.
```
Comment (katjam) [inline src/pages/Tap.vue]: double indent - but ok otherwise

File: src/store.js [hunk 4]
```diff
@@ -186,10 +186,11 @@ store.transitionVouchers = function (transition, vouchers, success, failure) {
         'vouchers': vouchers
     };
     this.netMgr.apiPost('vouchers', postData,
-        function () {
-            if (success) {success()}
-        }, function() {
-            if (failure) {failure()}
+        function (response) {
+            if (success) {success(response);}
+        }.bind(this),
```
Comment (katjam) [inline src/store.js]: don't need the bind anymore, nor the ; in the inline ifs.

**Review by katjam — APPROVED** [review]
Few little syntax tidys then :shipit:

File: src/pages/Payment.vue (hunk 1 above)
Comment (charlesstrange2) [reply src/pages/Payment.vue:87]: I think so, will JS throw a fit if it's passed a variable in a function when it's not expecting it?

File: src/pages/Payment.vue (hunk 2 above)
Comment (charlesstrange2) [reply src/pages/Payment.vue:91]: ditto above - transition is now feeding response and error objects to success/failure callbacks.

File: src/store.js (hunk 4 above)
Comment (charlesstrange2) [reply src/store.js]: okedoke

**Pull request merged**